Feature/implement claim module

Zlatonn opened the pull request:
**Features Included**
- Implement claim module
```
- Create claims service
- Get list claims service
- Get top claim user service
- Get top claim menu service
```
- Add pagination to menu service
```
- Get list menu service
```

**Preview Result**
- Create claims service

<img width="1464" height="822" alt="image" src="https://github.com/user-attachments/assets/49aad17e-8980-4c0f-86bc-c9681487d0c5" />

- Get list claims service

<img width="1474" height="819" alt="image" src="https://github.com/user-attachments/assets/ea52af34-ce0a-4158-bc9d-cd38e6d17999" />

- Get claims by user service

<img width="1484" height="828" alt="image" src="https://github.com/user-attachments/assets/961c6bf1-5f54-4d05-9e3d-ca2164f8cfe1" />

- Get top claim user service

<img width="1483" height="766" alt="image" src="https://github.com/user-attachments/assets/a04bcb36-72fa-4048-b5d3-1a17e52325ae" />

- Get top claim menu service

<img width="1485" height="824" alt="image" src="https://github.com/user-attachments/assets/78946b27-11f2-4b05-a774-ad51ef424de4" />

___

- Get list menu service

<img width="1483" height="824" alt="image" src="https://github.com/user-attachments/assets/5eac3e87-a3e4-4a1f-b965-7167bfd5a12f" />

Zlatonn committed:
implement claim menu

- create includes DTO
- create claim module
- create claim controllers
- create claim services
- create date helper
- apply claim module to app module

Zlatonn committed:
fix menu module

- add pagination for getListMenu service
- delete unused class-validator decorators

Zlatonn committed:
add date range filter to some claim services

- add pagination for getListMenu service
- delete unused class-validator decorators

looksocii commented on src/modules/claim/claim.service.ts:
```diff
@@ -0,0 +1,346 @@
+  async createClaims(id: number, req: CreateClaimDto) {
+    const { orders } = req;
+
+    const user = await this.prismaService.user.findUnique({
```
ไม่ต้องเช็คเพิ่มก็ได้ครับ เพราะ user id นี้ แกะออกมาจาก token
ถ้ามันไม่มีอยู่จริงจะ verify token ไม่ผ่านครับ

looksocii commented on src/modules/claim/claim.service.ts:
```diff
@@ -0,0 +1,346 @@
+      SELECT
+        users.id,
+        users.first_name,
+        users.last_name,
```
response เป็น camel case มั้ยครับ

looksocii commented on src/modules/claim/dtos/get-claim-by-user.dto.ts:
```diff
@@ -0,0 +1,41 @@
+export class GetClaimByUserDto {
+  @IsNotEmpty()
+  @Transform(({ value }) => getStartDate(value))
+  startDate: Date;
```
เพิ่ม IsDate มั้ย

Zlatonn committed:
request change#1

- add @IsDate decorator to date fields in Claim dtos
- delete double check user condition at createClaim service
- return sql result with camelCase format (using column alias)
